assembly/src/assembler/mod.rs

```diff
@@ -381,9 +385,18 @@ impl Assembler {
             .get_procedure(entrypoint)
             .expect("compilation succeeded but root not found in cache");
 
+        let (mast_forest, id_remappings) = dead_code_elimination(mast_forest_builder.build());
+        let entry_node_id = {
+            let old_entry_node_id = entry_procedure.body_node_id();
+
+            id_remappings
+                .map(|id_remappings| id_remappings[&old_entry_node_id])
+                .unwrap_or(old_entry_node_id)
+        };
```

Comment:
The procedure cache in `MastForestBuilder` stores procedures that were compiled against the old/unpruned `MastForest`. So all `MastNodeId`s stored in compiled procedures need to be remapped. Since in `assemble_library` we don't get a `MastNodeId` from a compiled procedure in the procedure cache, we don't need to do any remapping.